src/hotspot/cpu/ppc/globals_ppc.hpp

```diff
@@ -81,73 +81,74 @@ define_pd_global(intx, InitArrayShortSize, 9*BytesPerLong);
                                                                             \
-  product(bool, UseCountLeadingZerosInstructionsPPC64, true,                \
+  product(bool, UseCountLeadingZerosInstructionsPPC64, true, DIAGNOSTIC,    \
           "Use count leading zeros instructions.")                          \
                                                                             \
-  product(bool, UseCountTrailingZerosInstructionsPPC64, false,              \
+  product(bool, UseCountTrailingZerosInstructionsPPC64, false, DIAGNOSTIC,  \
           "Use count trailing zeros instructions.")                         \
                                                                             \
-  product(bool, UseExtendedLoadAndReserveInstructionsPPC64, false,          \
+  product(bool, UseExtendedLoadAndReserveInstructionsPPC64, false, DIAGNOSTIC,\
           "Use extended versions of load-and-reserve instructions.")        \
                                                                             \
-  product(bool, UseRotateAndMaskInstructionsPPC64, true,                    \
+  product(bool, UseRotateAndMaskInstructionsPPC64, true, DIAGNOSTIC,        \
           "Use rotate and mask instructions.")                              \
                                                                             \
-  product(bool, UseStaticBranchPredictionInCompareAndSwapPPC64, true,       \
-          "Use static branch prediction hints in CAS operations.")          \
-  product(bool, UseStaticBranchPredictionForUncommonPathsPPC64, false,      \
-          "Use static branch prediction hints for uncommon paths.")         \
-                                                                            \
   /* Trap based checks. */                                                  \
   /* Trap based checks use the ppc trap instructions to check certain */    \
   /* conditions. This instruction raises a SIGTRAP caught by the      */    \
   /* exception handler of the VM.                                     */    \
   product(bool, UseSIGTRAP, true,                                           \
           "Allow trap instructions that make use of SIGTRAP. Use this to "  \
           "switch off all optimizations requiring SIGTRAP.")                \
-  product(bool, TrapBasedICMissChecks, true,                                \
+  product(bool, TrapBasedICMissChecks, true, DIAGNOSTIC,                    \
           "Raise and handle SIGTRAP if inline cache miss detected.")        \
-  product(bool, TraceTraps, false, "Trace all traps the signal handler"     \
-          "handles.")                                                       \
                                                                             \
-  product(bool, ZapMemory, false, "Write 0x0101... to empty memory."        \
-          " Use this to ease debugging.")                                   \
```

Comment:
Future cleanup: this feels like it should be in shared code. The usual way to do this is to zap in DEBUG.

Reply:
Yeah, sounds like this functionality derserves some overwork. But I only want to make it diagnostic with this change.